added option to combine group layers into a single switcher entry

ATGardner opened the pull request:
Added an option to use a Group layer as a regular layer (radio button/check box), while not showing its sub-layers.

walkermatt commented:
Thanks, this would satisfy #22 and #35. Would you mind adding a test and a combined base map example (satellite with roads would be ideal)?

ATGardner commented:
Sure. I'll do it in the coming days.

ATGardner commented:
This current naive code change basically transforms a Group layer into a normal layer, which means it doesn't show any sub-layers it has, in the panel.

I have made some changes now, which removes the "combine" flag, and now any Group layer which has a "type" ("base"/"overlay") will show an input (radio/checkbox), and also render its sub-layers.

All the current tests pass, but I am thinking that maybe I should also make the sub-layers input element disabled, if the top layer is not currently checked. This will require more tests, and a bit more code. Should I try and implement such a feature? Are all the cases clearly defined (for example - a "base" group layer, with two "overlay" sub layers - what does it mean if it is selected, and neither sub-layer is checked? Or an "overlay" group layer, with a "base" sub layer - Can the map now have two base layers selected, one outside the group layer, and another inside it?)

walkermatt commented:
I'm in favour of keeping things simple, just adding an optional `combine` property that can be added to a group in order for it to be displayed as a layer.

ATGardner commented:
I've added two tests, and fixed the other tests to pass.
I still think that #37 suggestion might be simpler for users, though it will require a little change in the renderLayer_ function (similar to what was suggested there).

walkermatt commented:
@ATGardner thanks for your work on this, really appreciated :smiley_cat:
